components/num-util/src/digit_converter.rs

```diff
@@ -0,0 +1,101 @@
+use std::convert::TryInto;
+
+/// Converts between chars and u8 decimal digits using a zero char as a reference point.
+pub(crate) struct DigitConverter {
+    /// The char corresponding to the digit 0.
+    pub zero_char: char,
+}
+
+impl DigitConverter {
+    /// Converts from a decimal digit (which must be between 0 and 9) to a char.
+    pub(crate) fn char_for(&self, d: u8) -> char {
+        debug_assert_ge!(d, 0);
+        debug_assert_le!(d, 9);
+        let cp: u32 = self.zero_char as u32 + d as u32;
+        // TODO: Is there an unchecked version we can use here, if we check zero_char earlier?
+        cp.try_into().unwrap_or('\u{FFFD}')
+    }
+
+    /// Converts from a char to a decimal digit, or returns None if out of range.
+    #[allow(dead_code)]
+    pub(crate) fn checked_digit_for(&self, c: char) -> Option<u8> {
+        match (c as u32).checked_sub(self.zero_char as u32) {
+            Some(v) => {
+                if v <= 9 {
+                    Some(v as u8)
+                } else {
+                    None
+                }
+            }
+            None => None,
+        }
+    }
```

Comment:
how is it different from:

```rust
let b = c as u8;
if b >= b'0' && b <= b'9' {
    Some(c)
} else {
    None
}
```
?

Reply:
I think that doesn't work with non-ASCII code points?

There's a unit test below that shows how this works with other numbering systems.

Do you think we should use a different code path for ASCII digits for better efficiency?

Reply:
OK, I thought about it and right now there is only 1 call site for this class, and it can be replaced by a simple `b'0' + d`.  I removed DigitConverter from this PR and will add it back again in the future if it is needed.